src/plugin/helpers/element.ts

```diff
@@ -122,3 +121,24 @@ const getValuesFromContentNode = (node: Node, serializer: DOMSerializer) => {
   e.appendChild(dom);
   return e.innerHTML;
 };
+
+export const createElementDataValidator = <
+  FDesc extends FieldDescriptions<keyof FDesc>,
+  ElementNames extends keyof ESpecMap,
+  ExternalData,
+  ESpecMap extends ElementSpecMap<FDesc, ElementNames, ExternalData>
+>(
+  elementTypeMap: ESpecMap
+) => (
+  elementName: keyof ESpecMap,
+  values: unknown
```

Comment:
Torn on typesafety here. I think we do expect our data to be a valid shape for the element, even if it doesn't pass validation – that is to say, our validators expect an object that matches the type provided by the Element, even if the data itself is invalid.

Suspect that if keys are missing, for example, our validators will throw errors. So a lack of typesafety here could lead to runtime problems.

We could use a signature of the kind used in `insertElement` to make our `values` typesafe rather than `unknown`. I'd value your thoughts!

Reply:
Worked through this offline, it's now ready for re-review.